lib/modules/manager/devbox/artifacts.ts

```diff
@@ -0,0 +1,67 @@
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import type { ExecOptions } from '../../../util/exec/types';
+import { getSiblingFileName, readLocalFile } from '../../../util/fs';
+import { getRepoStatus } from '../../../util/git';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+
+export async function updateArtifacts(
+  updateConfig: UpdateArtifact,
+): Promise<UpdateArtifactsResult[] | null> {
+  const lockFileName = getSiblingFileName(
+    updateConfig.packageFileName,
+    'devbox.lock',
+  );
+  const existingLockFileContent = await readLocalFile(lockFileName, 'utf8');
+  if (!existingLockFileContent) {
+    logger.debug('No devbox.lock found');
+    return null;
+  }
+  const execOptions: ExecOptions = {
+    cwdFile: updateConfig.packageFileName,
+    toolConstraints: [
+      {
+        toolName: 'devbox',
+        constraint: updateConfig.config.constraints?.devbox,
+      },
+    ],
+    docker: {},
+    userConfiguredEnv: updateConfig.config.env,
+  };
+
+  let cmd = '';
+  if (updateConfig.config.isLockFileMaintenance) {
```

Comment:
you need an additional check

https://github.com/renovatebot/renovate/blob/3cae4d8c214a3205dadbd4e9bf2dc76f769521f7/lib/modules/manager/bun/artifacts.ts#L20-L25

Reply:
☝️ ❓ There are edge cases where `config.isLockFileMaintenance === false` but `config.updateType === 'lockFileMaintenance'`